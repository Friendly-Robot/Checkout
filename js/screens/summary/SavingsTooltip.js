/**
 * @flow
 */

"use strict";

import React, { useState } from "react";
import { findNodeHandle, StyleSheet, View } from "react-native";
import NativeMethodsMixin from 'NativeMethodsMixin';
import AppButton from "../../common/AppButton";
import AppColors from "../../common/AppColors";
import { Heading3 } from "../../common/AppText";
import DualItemRow from "../../common/DualItemRow";
import Tooltip from "../../common/Tooltip";

type props = {
  savings: Number,
};
const SavingsTooltip = ({
  savings = 0,
}: Props) => {

  const [showTooltip, setShowTooltip] = useState(false);
  const [tooltipBounds, setTooltipBounds] = useState({});

  let tooltipRef;

  return (
    <View
      onLayout={() => timeoutLayout(() => onTooltipLayout(tooltipRef, setTooltipBounds))}
      ref={(ref) => {tooltipRef = ref;}}
    >
      <DualItemRow>
        <AppButton
          caption={"Pickup savings"}
          captionStyle={{...styles.tooltipCaption, ...styles.underline}}
          hitSlop={5}
          onPress={() => setShowTooltip(!showTooltip)}
          theme={"none"}
          type={"none"}
        />
        <Heading3 style={[styles.bold, styles.red]}>{savings}</Heading3>
      </DualItemRow>

      {showTooltip &&
          <Tooltip
            bodyStyle={{ left: tooltipBounds.x, top: tooltipBounds.y,  ...styles.tooltipWrapper }}
            closeModal={() => setShowTooltip(!showTooltip)}
            content={"Picking up your order in the store helps cut costs, and we pass the savings on to you."}
            visible={showTooltip}
          />}
    </View>
  )
}

const timeoutLayout = cb => setTimeout(cb, 0);

const onTooltipLayout = (ref, setTooltipBounds) => {
  if (ref) {
    NativeMethodsMixin.measureInWindow.call(
      findNodeHandle(ref),
      (x, y) => setTooltipBounds({ x, y })
    );
  }
}

const styles = StyleSheet.create({
  bold: {
    fontWeight: "500"
  },
  red: {
    color: AppColors.red
  },
  tooltipCaption: {
    color: AppColors.black
  },
  tooltipWrapper: {
    marginRight: 150,
    marginTop: 45
  },
  underline: {
    textDecorationLine: "underline"
  }
});

export default SavingsTooltip;
