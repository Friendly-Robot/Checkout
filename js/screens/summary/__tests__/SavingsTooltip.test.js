/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SavingsTooltip from '../SavingsTooltip';


describe('<SavingsTooltip />', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = {
      handlePromoSubmit: () => {},
    };

    wrapper = shallow(<SavingsTooltip {...props} />);
  })

  it('should be defined', () => {
    expect(SavingsTooltip).toBeDefined();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<SavingsTooltip {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
