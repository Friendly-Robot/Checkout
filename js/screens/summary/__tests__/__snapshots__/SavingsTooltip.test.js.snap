// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavingsTooltip /> renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "marginRight": 10,
          },
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        hitSlop={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 52,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 26,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "padding": 0,
              },
              Object {
                "borderRadius": 0,
                "justifyContent": "flex-start",
                "paddingHorizontal": 0,
              },
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(3, 3, 3, 1)",
                },
                Array [
                  Object {
                    "fontSize": 15,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 114, 206, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  undefined,
                  Object {
                    "color": "rgba(3, 3, 3, 1)",
                    "textDecorationLine": "underline",
                  },
                ],
              ]
            }
          >
            Pickup savings
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "marginLeft": 10,
          },
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(3, 3, 3, 1)",
              "fontSize": 17,
            },
            Array [
              Object {
                "fontWeight": "500",
              },
              Object {
                "color": "rgba(240, 52, 52, 1)",
              },
            ],
          ]
        }
      >
        -$0
      </Text>
    </View>
  </View>
</View>
`;
